src/main/java/com/ibm/crypto/plus/provider/RSASignature.java

```diff
@@ -47,6 +49,23 @@ protected void engineInitVerify(PublicKey publicKey) throws InvalidKeyException
             throw new InvalidKeyException("Key is not an RSAPublicKey");
         }
 
+        try {
+            if (provider.isFIPS()) {
+                RSAKeyFactory.checkKeyLengths(((java.security.interfaces.RSAPublicKey) publicKey).getModulus().bitLength(),
+                        RSAKeyGenParameterSpec.F4,
+                        RSAKeyFactory.MIN_MODLEN_FIPS_PUB,
+                        64 * 1024,
+                        RSAKeyFactory.ALLOWABLE_MODLEN_FIPS_VERIFY);
+            } else {
+                RSAKeyFactory.checkKeyLengths(((java.security.interfaces.RSAPublicKey) publicKey).getModulus().bitLength(),
+                        RSAKeyGenParameterSpec.F4,
+                        RSAKeyFactory.MIN_MODLEN_NONFIPS,
+                        64 * 1024);
+            }
+
+        } catch (InvalidKeyException e) {
+            throw new InvalidParameterException(e.getMessage());
```

Comment:
Please add a caused by to this exception by passing the original exception to the newly thrown exception.

Reply:
this should be throw the `e` exception directly.

Reply:
Seems like we can get rid of the entire try catch block here as any `InvalidKeyException` can be directly thrown from this method correct?

Reply:
right